src/interpreter/mod.rs

```diff
@@ -136,10 +133,28 @@ enum TerminatorTarget {
 
 #[derive(Clone, Debug, Eq, PartialEq, Hash)]
 enum ConstantId<'tcx> {
```

Comment:
Perhaps make a `struct` that always has `DefId` and `Substs` and avoid the need for the methods.